src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateArray = [];
+  let newState = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'clear':
+        newState = {};
+        break;
+
+      case 'addProperties':
+        Object.assign(newState, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete newState[key];
+        }
+
+        break;
+
+      default:
+        break;
+    }
```

Comment:
Consider using a `switch` statement default case for error handling. If an action type is passed that is not recognized, it would be helpful to throw an error or handle this case explicitly.